proc/proc_darwin.c

```diff
@@ -100,10 +130,15 @@ get_threads(task_t task, void *slice, int limit) {
 }
 
 int
-thread_count(task_t task) {
+thread_count(int pid) {
 	kern_return_t kret;
 	thread_act_array_t list;
 	mach_msg_type_number_t count;
+	mach_port_t self = mach_task_self();
+	task_t task;
+
+	kret = task_for_pid(self, pid, &task);
```

Comment:
Calling `task_for_pid` is likely going to be pretty inefficient. Since we now know the error lies with OSX changes tasks underneath us during exec, can we just use the task we've reaquired?